crates/turbopack-cli/src/build/mod.rs

```diff
@@ -0,0 +1,322 @@
+
+    let entries = entry_requests
+        .into_iter()
+        .map(|request_vc| async move {
+            let ty = Value::new(ReferenceType::Entry(EntryReferenceSubType::Undefined));
+            let request = request_vc.await?;
+            Ok(*origin
+                .resolve_asset(request_vc, origin.resolve_options(ty.clone()), ty)
+                .primary_assets()
+                .await?
+                .first()
+                .with_context(|| {
+                    format!(
+                        "Unable to resolve entry {} from directory {}.",
+                        request.request().unwrap(),
+                        project_dir
+                    )
+                })?)
+        })
+        .try_join()
+        .await?;
+
+    let modules = entries.into_iter().map(|entry| {
+        context.process(
+            entry,
+            Value::new(ReferenceType::Entry(EntryReferenceSubType::Undefined)),
+        )
+    });
+
+    let entry_chunk_groups = modules
+        .map(|entry_module| async move {
+            Ok(
+                if let Some(ecmascript) =
+                    EcmascriptModuleAssetVc::resolve_from(entry_module).await?
+                {
+                    AssetsVc::cell(vec![BuildChunkingContextVc::resolve_from(chunking_context)
+                        .await?
+                        .unwrap()
+                        .generate_entry_chunk(
+                            build_output_root
+                                .join(
+                                    entry_module
+                                        .ident()
+                                        .path()
+                                        .file_stem()
+                                        .await?
+                                        .as_deref()
+                                        .unwrap(),
+                                )
+                                .with_extension("entry.js"),
+                            ecmascript.into(),
+                            EvaluatableAssetsVc::one(ecmascript.into()),
+                        )])
+                } else if let Some(chunkable) =
+                    ChunkableModuleVc::resolve_from(entry_module).await?
+                {
+                    chunking_context.chunk_group(chunkable.as_root_chunk(chunking_context))
+                } else {
+                    // TODO convert into a serve-able asset
+                    bail!(
+                        "Entry module is not chunkable, so it can't be used to bootstrap the \
+                         application"
+                    )
+                },
+            )
+        })
+        .try_join()
+        .await?;
+
+    for chunk_group in entry_chunk_groups {
+        for entry in &*chunk_group.await? {
+            for asset in &*all_assets_from_entry(entry.to_owned()).await? {
+                asset.content().write(asset.ident().path()).await?;
+            }
+        }
+    }
```

Comment:
That's pretty inefficient since it's generating and writing all assets sequentially. You could generate and write them all in parallel.